File: System Design Java Codes/Behavioral Design Pattern/Observer_Design_Pattern/Without/WeatherStation.java
package WithoutObserverPattern;

class WeatherStation {
    public String weather;
    private PhoneDisplay phoneDisplay;
    private TVDisplay tvDisplay;

    public WeatherStation(PhoneDisplay phoneDisplay, TVDisplay tvDisplay) {
        this.phoneDisplay = phoneDisplay;
        this.tvDisplay = tvDisplay;
    }

    public void setWeather(String newWeather) {
        this.weather = newWeather;
        phoneDisplay.update(newWeather);
        tvDisplay.update(newWeather);
    }
}

File: System Design Python Codes/Behavioral Design Pattern/Observer_Design_Pattern/Without/WeatherStation.py

import PhoneDisplay
import TVDisplay

class WeatherStation:   
    def __init__(self, phoneDisplay, tvDisplay):
        self.phoneDisplay = phoneDisplay
        self.tvDisplay = tvDisplay
        
    def setWeather(self, weather):
        self.phoneDisplay.update(weather)
        self.tvDisplay.update(weather)